sklearn/metrics/_pairwise_distances_reduction/_dispatcher.py

```diff
@@ -119,26 +119,7 @@ def is_valid_sparse_matrix(X):
             and metric in cls.valid_metrics()
         )
 
-        # The other joblib-based back-end might be more efficient on fused sparse-dense
-        # datasets' pairs on metric="(sq)euclidean" for some configurations because it
-        # uses the Squared Euclidean matrix decomposition, i.e.:
-        #
-        #       ||X_c_i - Y_c_j||² = ||X_c_i||² - 2 X_c_i.Y_c_j^T + ||Y_c_j||²
```

Comment:
Is this "trick" still documented somewhere in the pairwise distances codebase?

Reply:
Yes, it is given here:

https://github.com/scikit-learn/scikit-learn/blob/dfe9e2e118bd40fa4cd3fce7e007bc7bf51c9033/sklearn/metrics/_pairwise_distances_reduction/_middle_term_computer.pyx.tp#L77-L93

I am thinking of writing a couple of notes for the design of `sklearn.metrics._pairwise_distances_reduction`. What do you think of it?

Reply:
That would be great. Not too many details, but giving a good overview.